datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategy.java

```diff
@@ -470,6 +499,24 @@ protected int constructExpectedNumberOfTasks(DatastreamGroup dg, List<String> in
     return expectedNumberOfTasks;
   }
 
+  private void updateOrRegisterElasticTaskAssignmentMetrics(DatastreamGroupPartitionsMetadata datastreamPartitions,
+      int totalTaskCount) {
+    int totalPartitions = datastreamPartitions.getPartitions().size();
+    int actualPartitionsPerTask = (totalTaskCount / totalPartitions)
```

Comment:
`totalTaskCount` and `totalPartitions` need to be flipped, otherwise `actualPartitionsPerTask` will always resolve to 0 or 1.